### PR TITLE
Revert "Revert "qa_crowbarsetup: use crowbarctl to commit proposals""

### PR DESCRIPTION
**Pull request by rsalevsky**
This PR is important for SAP gating.

### CODE REVIEW

**Comment by jdsn** [comment]
Another reason for this change is, that we were not able to change the timeout for the `crowbar` command, but we now can change it (and have to due to longer requests with bigger clouds for eg. SAP).

**Comment by vuntz** [comment]
Can you run a HA job to make sure 10 minutes is fine? I think we have a higher timeout in the deprecated CLI.

**Comment by rsalevsky** [comment]
@vuntz It's 500 in the old CLI.

**Comment by jdsn** [comment]
@vuntz sure I can

**Comment by vuntz** [comment]
> @vuntz It's 500 in the old CLI.

In the library yes, but all barclamp-specific subcommands overwrite this to 3600 ;-)

**Comment by jdsn** [comment]
ha build is triggered but the build queue for ha jobs is quite long currently - so it will take some time until it can start

**Comment by jdsn** [comment]
@vuntz ... but all barclamp-specific subcommands overwrite this to 3600 ;-)

Then why do we have a timeout (smaller than two hours) at all? mkcloud builds will be killed after two hours of inactivity anyway.

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -98,6 +98,8 @@ export want_postgresql=${want_postgresql:-1}
 if iscloudver 6plus ; then
     export CROWBAR_EXPERIMENTAL=true
     export CROWBAR_VERIFY_SSL=false
+    # set timeout to 10min to avoid bsc#988030
+    export CROWBAR_TIMEOUT=600
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: can we use the old timeout of 3600 here or why does SAP require a lower timeout?

**Comment by rsalevsky** [comment]
> In the library yes, but all barclamp-specific subcommands overwrite this to 3600 ;-)

@vuntz Now I'm confused, crowbar (puma) would not keep the connection open for such a long time. Can you point me to the code which is setting the timeout to 3600?

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: @dirkmueller thanks for rereading my comment.

I said that we need to change it for SAP to be longer than the old timeout. We have builds of SAP PRs where a timeout hits after about 5 minutes. That is too short, so we need to be able to raise this.

What the the old value is (or values are) is currently discussed here.

**Comment by vuntz** [comment]
> @vuntz Now I'm confused, crowbar (puma) would not keep the connection open for such a long time. Can you point me to the code which is setting the timeout to 3600?

@rsalevsky https://github.com/crowbar/crowbar-openstack/blob/master/bin/crowbar_nova#L21

**Comment by rsalevsky** [comment]
Ok, now we have a lot of different timeouts. 

apache = 25min
puma = 60min
old CLI = 60min
new CLI = 5min
new CLI in mkcloud = 10min

How do we unify this timeout now? Should we go with 60min or the current maximum of 25min?

**Comment by jdsn** [comment]
@rsalevsky thanks for summarizing this

IMHO we should do two things:
1. align the timeouts in apache and puma - having them different does not make sense
2. set the default in crowbarctl to the same (or to the mininum of)  as apache and puma

The long timeouts in apache and puma are there for a reason (I assume). Making crowbarctl useless in many cases (too big clouds) due to a too low timeout setting does not make any sense.

So my proposal is: set all to 60min

@vuntz @dirkmueller whats your take on this?

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: @jdsn the old timeout before the change yesterday was client side 1 hour and server side 25 min. where is the 5 min coming from?

**Comment by vuntz** [comment]
I'd go for 60 minutes for now; this is probably too much, but I don't think it's that much of a big deal.

**Comment by dirkmueller** [comment]
@jdsn I don't see the client side timeout and the server side timeout strongly coupled. for me it makes sense that the client side timeout is higher than the server side. I agree that there is an inconsistency server side that needs to be fixed. 

for the client side I suggest to change the default (maybe even in the code directly) to the old default of 1 hour. hacking that around in the qa_crowbarsetup.sh automation looks like the wrong place to me.

**Comment by rsalevsky** [comment]
Ok, I take this as agreement to sync all timeouts to 60min directly in the code. I will open all required PRs to do so.

**Comment by jdsn** [comment]
> I don't see the client side timeout and the server side timeout strongly coupled
>
> for the client side I suggest to change the default [...] to the old default of 1 hour

lol

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: > before the change [...]

right, that's what we know now - we didn't knew about the mess before
So its good that we found this during this discussion.

> And why are you saying that raising to 1 hour as I suggested would not solve the problem?

I didn't say that. I even proposed the same :)  But I would also make it consistent.

Let me repeat the history for you:

- We saw mkcloud builds of SAP PRs failing in crowbar commit due to timeouts.
- We could not change the timeout for the crowbar command
- As the crowbar command is obsolete anyway, we had the idea to switch to crowbarctl (for now only for proposal commit), as we can change the timeout from mkcloud via env variable
- then this PR was created
- and then during the discussion we found out about the configuration mess

**Comment by dirkmueller** [comment]
> lol

Not sure whats funny about that. I see zero proof that the SAP ci timed out on a *client* timeout of 5 minutes, and before the change that got reverted, the client timeout was 1 hour.

**Comment by rsalevsky** [comment]
Here are the fixes:

https://github.com/crowbar/crowbar-core/pull/1206
https://github.com/crowbar/crowbar-init/pull/74
https://github.com/crowbar/crowbar-client/pull/161

After this the timeout is always the same and we can merge this PR.

**Comment by dirkmueller** [comment]
@rsalevsky we still need a new crowbarctl release for C6+C7 before we can merge this patch.

**Comment by rsalevsky** [comment]
@dirkmueller I updated the gem already in our internal packages.

**Comment by rsalevsky** [comment]
HA gating failure looks unrelated.

**Comment by bmwiedemann** [comment]
https://ci.suse.de/job/openstack-mkcloud/59472/ has a timeout after 20 minutes, even though cinder never took this long:
```
14:47:35 +(qa_crowbarsetup.sh:3201) update_one_proposal(): crowbar cinder proposal commit default
15:07:44 Failed to talk to service proposal commit: 500: {"status":"500","error":"Internal Server Error"}
```

would that be better or worse with this patch?

**Comment by bmwiedemann** [comment]
apart from this, the crowbarctl timeout should be less than 1h, because jenkins will kill jobs the hard way after 1h and you will not get supportconfigs etc to help you debug.

**Comment by dirkmueller** [comment]
well, this is a regression then - previously the server timed out after 25 min. 

personally, I think it would be better to have a shorter timeout in the server side than in the client side, so that the client actually has a chance of catchign the server timeout and do something appropriate.

**Comment by rsalevsky** [comment]
All depended PRs are merged, can we merge this one as well?

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -3173,6 +3173,18 @@ function set_noproxyvar
     no_proxy="${no_proxy%,}";
 }
 
+# commit a proposal, but use crowbarctl from cloud6 on
+function crowbar_proposal_commit
+{
+    local proposal="$1"
+    local proposaltype="${2:-default}"
+    if iscloudver 6plus ; then
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: We only want to run this if crowbarctl is new enough.. so GM6, GM6+up and so on are currently not good?

**Comment by dirkmueller** [comment]
Did the crowbarctl update get released everywhere?

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (rsalevsky) [inline scripts/qa_crowbarsetup.sh]: Good point, the updates are not out yet (they go together with the upgrade update). Should we set the timeout env variable to also cover the GM6 & GM7 case?

**Comment by rsalevsky** [comment]
Re-added the second commit to fix the timeout issue for GM.

**Comment by bmwiedemann** [comment]
Test with HA+SSL and applying all proposals a 2nd time passed. With old crowbar CLI, it triggered the bug reliably for me.

**Pull request merged**